src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -228,15 +253,15 @@ CHIP_ERROR OTARequestor::SetupCASESessionManager(chip::FabricIndex fabricIndex)
     // CSM has not been setup so create a new instance of it
     if (mCASESessionManager == nullptr)
     {
-        chip::Server * server         = &(chip::Server::GetInstance());
-        chip::FabricInfo * fabricInfo = server->GetFabricTable().FindFabricWithIndex(fabricIndex);
+        Server * server         = &(chip::Server::GetInstance());
```

Comment:
Could this have been passed/set as an constructor/init arg? Otherwise we are tying a singleton to this, which means there is no way to ever do two independant cluster servers in same process if OTA is to be supportable.

Reply:
Just wanted to clarify, you'd like server to be set/stored as a member variable to the OTARequestor class? From the caller, it could still pass in the singleton server but from an OTARequestor perspective, it just knows it has an instance of server. Does that sound right?